core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -147,7 +160,9 @@ class BrokerMetadataListener(
 
   private def maybeStartSnapshot(): Unit = {
     snapshotter.foreach { snapshotter =>
-      if (snapshotter.maybeStartSnapshot(_highestTimestamp, _delta.apply())) {
+      if (metadataFaultOccurred.get()) {
+        trace("Not starting metadata snapshot since we previously had an error")
```

Comment:
I went with TRACE here to avoid spamming the logs with this message on each metadata commit

Reply:
Honestly I think this should be WARN or ERROR to draw our attention to it.

Reply:
My concern is that in an emergency scenario when things are going badly, we could lose important logs due to log rotation if we're spamming a bunch of the same error/warn messages.

Reply:
Yeah. That's fair.

I guess it's OK like it is for now. I am going to refactor this as part of the effort to unify Image handling between broker + controller.